api/v1alpha1/gatewayclass_types.go

```diff
@@ -26,22 +26,95 @@ import (
 // +k8s:openapi-gen=true
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 
-// GatewayClass describes a class of gateways av
-// for defining access to their routed services. GatewayClass allow a
+// GatewayClass describes a class of gateways available to the user
+// for creating Gateway resources.
 //
-// GatewayClass is a non-namespaced resource.
+// GatewayClass is a Cluster level resource.
+//
+// Support: Core.
 type GatewayClass struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
-	// Controller is a domain/path that denotes which controller is responsible
-	// for this class. Example: "acme.io/gateway-controller".
-	Controller string `json:"controller" protobuf:"bytes,2,opt,name=controller"`
+	// Spec for this GatewayClass.
+	Spec GatewayClassSpec `json:"spec,omitempty"`
+	// Status of the GatewayClass.
+	Status GatewayClassStatus `json:"status,omitempty"`
+}
+
+// GatewayClassSpec reflects the configuration of a class of Gateways.
+type GatewayClassSpec struct {
+	// Controller is a domain/path string that indicates the
+	// controller is responsible for this class. Example:
+	// "acme.io/gateway-controller".
+	//
+	// This will be used to distinguish which controller should
+	// manage Gateways of this class.
+	//
+	// Support: Core
+	//
+	// +required
+	Controller string `json:"controller"`
+
 	// Parameters is a controller specific resource containing the
-	// configuration parameters corresponding to this class. This is optional
-	// if the controllers does not require any additional configuration.
+	// configuration parameters corresponding to this class. This
+	// is optional if the controllers does not require any
+	// additional configuration.
+	//
+	// Valid types for reference is entirely up to the Controller
+	// that is managing Gateways of this class. It can be a
+	// core.ConfigMap or a custom resource.
+	//
+	// Support: Custom
+	//
+	// +optional
+	Parameters *core.ObjectReference `json:"parameters,omitempty" protobuf:"bytes,3,opt,name=parameters"`
```

Comment:
Maybe a separate PR, but maybe this field should be called "ParametersRef"

https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#object-references

Reply:
Thanks -- fixing.